### PR TITLE
Add API endpt. for sonos testing & cleaner code

### LINKED ISSUE #296: API endpoint for triggering individual SONOS Speakers
This feature will allow creating a 'TEST' feature on the setup app, where the user can identify the speaker being configured using a short audio feedback.

### PR DESCRIPTION
**Pull request by neelotpalnag**
Added API endpoint for Sonos test. However, the API is currently playing a file hosted on `files.senic.com`, but we intend to self-host the audio file on the hub itself.
This modification will be added in commits to the branch. 
Closes #296

### CODE REVIEW

File: senic_hub/backend/views/nuimo_components.py [hunk 1]
```diff
@@ -278,17 +280,20 @@ def get_test_response(request):
 
     component_type = component['type']
     component_ip = component['ip_address']
-    component_username = component['username']
+    try:
+        component_username = component['username']
+    except:  # pragma: no cover
+        pass
```
Comment (rudolfs) [inline senic_hub/backend/views/nuimo_components.py]: `component_username = component.get('username', None)`

File: senic_hub/backend/views/nuimo_components.py [hunk 2]
```diff
@@ -338,3 +359,25 @@ def test_blink_phue(component_ip, component_username, id):  # pragma: no cover
...
+        test_speaker.volume = 10
+        test_speaker.play_uri(uri_test)
+        test_speaker.get_current_track_info()
+        time.sleep(2)
+        test_speaker.stop()
+        try:
+            test_speaker.play_from_queue(0)
+            test_speaker.stop()
+        except:
+            pass
```
Comment (rudolfs) [inline senic_hub/backend/views/nuimo_components.py]: Generally swallowing all exceptions is not a good practice. Either handle a specific exception by specifying the exception class or if you do a catch all, log it somewhere, so we can debug later.

File: senic_hub/backend/views/nuimo_components.py [hunk 3]
```diff
@@ -300,8 +305,24 @@ def get_test_response(request):
                 'message': 'ERROR_PHUE_PUT_REQUEST_FAIL'
             }
 
-    # TODO : Add short audio test for sonos
-    # if component_type == 'sonos':
+    if component_type == "sonos":  # pragma: no cover
+        test_result = test_ring_sonos(component_ip)
```
Comment (rudolfs) [inline senic_hub/backend/views/nuimo_components.py]: Would make more sense if the method returned either `True` or `False`

**Comment by neelsenic** [comment]
Please ignore the previous THREE commits. They have conflicts with master & probably messed up by some git-quantum entanglement on my computer

**Comment by neelsenic** [comment]
**TODO: The hub's IP address must be updated in the (new) `hub_ip_address` field in production.ini everytime the hub is onboarded or is assigned a new IP under DHCP  
The probable implementation is in `senic-hub/senic_hub/backend/netwatch.py` . "Probable" .
Also, Travis check fails if `python-networkmanager` package is added to `senic-hub/setup.py` (or in `tox.ini`) :cry:

**Pull request closed without merging**